src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -69,8 +68,8 @@ public void setTags(Set<Tag> tags) {
         this.tags.setTags(tags);
     }
 
-    public void setLifeInsurances(Map<String, ? extends ReadOnlyInsurance> insurances) {
-        this.lifeInsuranceMap = (HashMap<String, LifeInsurance>) insurances;
+    public void setLifeInsurances(Map<UUID, ReadOnlyInsurance> insurances) throws DuplicateInsuranceException {
+        this.lifeInsuranceMap.setInsurances((insurances));
```

Comment:
double brackets? surprising travis didnt complain about this

Reply:
I'm surprised, too. Will fix it!